agent/lib/kontena/load_balancers/configurer.rb

```diff
@@ -79,26 +74,8 @@ def ensure_config(container)
       error exc.backtrace.join("\n") if exc.backtrace
     end
 
-    # @param [Docker::Container] container
-    def remove_config(container)
-      name = container.labels['io.kontena.load_balancer.name']
-      service_name = container.service_name_for_lb
-      mode = container.env_hash['KONTENA_LB_MODE'] || 'http'
-      info "un-registering #{service_name} from load balancer #{name} (#{mode})"
-      if mode == 'http'
-        etcd_path = "#{ETCD_PREFIX}/#{name}/services/#{service_name}"
-      else
-        etcd_path = "#{ETCD_PREFIX}/#{name}/tcp-services/#{service_name}"
-      end
-      rmdir(etcd_path)
-    rescue => exc
-      error "#{exc.class.name}: #{exc.message}"
-    end
-
-
-    # @param [Docker::Container] container
-    def remove_service(container)
-      service_name = container.service_name_for_lb
+    # @param [String] service_name
+    def remove_config(service_name)
```

Comment:
BTW this function will behave hilarously if `service_name = nil`... it will wipe out all LB services from etcd :P